File: .tmp_py4013/batch_02_sources/2219_maximum_sum_score_of_array.js
// LeetCode 2219 - Maximum Sum Score of Array
// https://leetcode.com/problems/maximum-sum-score-of-array/

/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumSumScore = function(nums) {
    let total = 0, pref = 0;
    for (const x of nums) total += x;
    let ans = -Infinity;
    for (const x of nums) {
        pref += x;
        ans = Math.max(ans, Math.max(pref, total - pref + x));
    }
    return ans;
};

File: .tmp_sc4013/java_06/2219_maximum_sum_score_of_array.java
// LeetCode 2219 - Maximum Sum Score of Array
// https://leetcode.com/problems/maximum-sum-score-of-array/

class Solution {
    public long maximumSumScore(int[] nums) {
        long total = 0, pref = 0;
        for (int x : nums) total += x;
        long ans = Long.MIN_VALUE;
        for (int x : nums) {
            pref += x;
            ans = Math.max(ans, Math.max(pref, total - pref + x));
        }
        return ans;
    }
}
